web/themes/contrib/civictheme/includes/form_element.inc

```diff
@@ -0,0 +1,425 @@
+      'id' => Html::cleanCssIdentifier($element['#id'] . '-' . $value),
+      'is_checked' => isset($element['#default_value']) ? (is_array($element['#default_value']) ? in_array($value, $element['#default_value']) : $element['#default_value'] == $value) : FALSE,
+      'is_disabled' => isset($element['#attributes']['disabled']),
+      'attributes' => $attributes,
+      'modifier_class' => $modifier_class,
+    ];
+  }
+
+  $variables['control'] = $control;
+
+  $is_horizontal = isset($element['#options_display']) && $element['#options_display'] == 'inline';
+  $is_inline = isset($element['#options_display']) && ($element['#options_display'] == 'side_by_side' || $element['#options_display'] == 'buttons_horizontal');
+
+  $variables['orientation'] = $is_horizontal ? 'horizontal' : 'vertical';
+  $variables['is_inline'] = $is_inline;
+
+  $variables['type'] = 'radio';
+}
+
+/**
+ * Implements template_preprocess_fieldset().
+ */
+function civictheme_preprocess_fieldset(array &$variables): void {
+  $element = $variables['element'];
+
+  if ($element['#type'] !== 'fieldset') {
+    return;
+  }
+
+  _civictheme_preprocess_form_element__theme($variables);
+  _civictheme_preprocess_form_element__classes($variables);
+  _civictheme_preprocess_form_element__description($variables);
+}
+
+/**
+ * Implements template_preprocess_input().
+ */
+function civictheme_preprocess_input__submit(array &$variables): void {
+  _civictheme_preprocess_form_element__theme($variables);
+
+  $type = 'primary';
+  if (is_array($variables['attributes']['class']) && in_array('secondary', $variables['attributes']['class'])) {
+    $type = 'secondary';
+  }
+  $variables['type'] = $type;
+
+  $variables['value'] = $variables['attributes']['value'] ?? t('Submit');
+}
+
+/**
+ * Preprocesses a generic form element.
+ */
+function _civictheme_preprocess_form_element__generic(array &$variables): void {
+  $element = $variables['element'];
+  $variables['label'] = isset($element['#title']) ? $element['#title'] : '';
+  $variables['type'] = isset($element['#type']) ? $element['#type'] : 'input';
+  $variables['orientation'] = isset($variables['orientation']) ? $variables['orientation'] : 'vertical';
+  $variables['is_required'] = isset($element['#required']) ? $element['#required'] : FALSE;
+  $variables['is_disabled'] = isset($element['#attributes']['disabled']);
+  $variables['description'] = isset($element['#description']) ? $element['#description'] : '';
+  $variables['message'] = isset($element['#errors']) ? $element['#errors'] : '';
+  $variables['is_invalid'] = isset($element['#errors']) && !empty($element['#errors']);
+}
+
+/**
+ * Preprocesses a form element control.
+ *
+ * @SuppressWarnings(PHPMD.CyclomaticComplexity)
+ * @SuppressWarnings(PHPMD.NPathComplexity)
+ */
+function _civictheme_preprocess_form_element__control(array &$variables): void {
+  $element = $variables['element'];
+
+  $allowed_attributes = ['size', 'maxlength', 'autocomplete'];
```

Comment:
List of other attributes allowed in render elements:
- open
- placeholder
- -id-
- aria-describedby
- readonly
- data-media-embed-host-entity-langcode
- data-drupal-settingstray
- aria-labelledby
- role
- aria-label
- data-drupal-selector
- formnovalidate
- pattern
- type
- step
- min
- max
- multiple

Reply:
My concern with this approach is we seem to be wresting drupal's control of the forms from modules to the theme layer.
Maybe this all just works with the form api but I think it needs to be tested with example form pages.
For example: \Drupal\Core\Render\Element\Button::getInfo
this provides process handlers including ajax. 
How does these information added to elements interact with CivicTheme?